langlib/langlib-test/src/test/resources/test-src/xmllib_constrained_test.bal

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'xml;
+
+public function basicXMLConstrainedType() {
+    'xml:Element elem1 = xml `<hello>xml content</hello>`;
+    xml<'xml:Element> elementSequence = xml `<hello>xml content</hello>`;
+    'xml:Element refer = elementSequence[0];
```

Comment:
Consider this:
```ballerina
public function main() {
    'xml:Element elem1 = xml `<hello>xml content</hello>`;
    xml<'xml:Element> elementSequence = xml `<hello>xml content</hello>`;
    'xml:Element refer = elementSequence[0];
    'xml:Element two = elementSequence[1]; // returns empty xml sequence. https://ballerina.io/ballerina-spec/spec.html#section_6.15
    io:println(two);
    anydata k = two;
    io:println(k is 'xml:Element);
    io:println(k is xml);
}
```

I think we made a mistake type checking xml member access expressions. In the line with a comment, out of range member acess returns empty xml value (different from list). So the lhs type should be `xml` or 'xml<element>` (to allow empty sequence).

Reply:
This is fixed via
https://github.com/ballerina-platform/ballerina-lang/pull/22323/commits/408178a5c64a763f7442b23de22fb83198e100bb

It also fixes the ability to cast empty sequence for text type.